### ISSUE TITLE
Using a listen with salt-ssh doesn't work, but watch does

### ISSUE THREAD

**Issue by edgan** [saltstack/salt#39239]
### Description of Issue/Question
I have a service that I check to see if pkgs, a directory, and a file change. If any change I want to restart the service. I am doing this with salt-ssh. Using a watch works, but using a listen doesn't.

I wanted to switch to listen to avoid the order being changed.

### Setup
Example: 
```
{% from 'foo/map.jinja' import foo with context %}
 
foo_service:
  module.wait:
    - name: service.systemctl_reload
    - listen:
      - pkg: foo_pkgs
      - file: foo_init_file
  service.running:
    - name: {{ foo.service.name }}
    - enable: True
    - listen:
      - pkg: foo_pkgs
      - file: foo_config_dir
      - file: foo_init_file
 ```
 
Output: 
16:38:16   Name: foo_pkgs - Function: pkg.installed - Result: Changed Started: - 16:37:42.563045 Duration: 13448.689 ms
16:38:16   Name: foo - Function: service.running - Result: Clean Started: - 16:37:57.601214 Duration: 21.391 ms

### Steps to Reproduce Issue
(Include debug logs if possible and relevant.)

### Versions Report
2016.11.1

**Comment by gtmanfred** [comment]
Thanks for reporting this, I have confirmed that this is not working with salt-ssh.

We will get it fixed.

Thanks,
Daniel

**Comment by gtmanfred** [comment]
```
[root@salt ~]# cat /srv/salt/test.sls
nginx:
  pkg.installed: []
  service.running:
    - listen:
      - pkg: nginx
      - file: nginx
  file.absent:
    - name: /etc/nginx/nginx.conf.default
```

Here it is with regular salt

```
[root@salt ~]# salt-call state.apply test
local:
----------
          ID: nginx
    Function: pkg.installed
      Result: True
     Comment: The following packages were installed/updated: nginx
     Started: 14:56:04.443512
    Duration: 8847.352 ms
     Changes:
              ----------
              nginx:
                  ----------
                  new:
                      1:1.10.2-1.el7
                  old:
              nginx-all-modules:
                  ----------
                  new:
                      1:1.10.2-1.el7
                  old:
              nginx-mod-http-geoip:
                  ----------
                  new:
                      1:1.10.2-1.el7
                  old:
              nginx-mod-http-image-filter:
                  ----------
                  new:
                      1:1.10.2-1.el7
                  old:
              nginx-mod-http-perl:
                  ----------
                  new:
                      1:1.10.2-1.el7
                  old:
              nginx-mod-http-xslt-filter:
                  ----------
                  new:
                      1:1.10.2-1.el7
                  old:
              nginx-mod-mail:
                  ----------
                  new:
                      1:1.10.2-1.el7
                  old:
              nginx-mod-stream:
                  ----------
                  new:
                      1:1.10.2-1.el7
                  old:
----------
          ID: nginx
    Function: service.running
      Result: True
     Comment: Started Service nginx
     Started: 14:56:13.721804
    Duration: 170.236 ms
     Changes:
              ----------
              nginx:
                  True
----------
          ID: nginx
    Function: file.absent
        Name: /etc/nginx/nginx.conf.default
      Result: True
     Comment: Removed file /etc/nginx/nginx.conf.default
     Started: 14:56:13.894335
    Duration: 0.932 ms
     Changes:
              ----------
              removed:
                  /etc/nginx/nginx.conf.default
----------
          ID: listener_nginx
    Function: service.mod_watch
        Name: nginx
      Result: True
     Comment: Service restarted
     Started: 14:56:13.895488
    Duration: 100.515 ms
     Changes:
              ----------
              nginx:
                  True

Summary for local
------------
Succeeded: 4 (changed=4)
Failed:    0
------------
Total states run:     4
Total run time:   9.119 s
```

and here it is with salt-ssh

```
[root@salt ~]# salt-ssh \* state.apply test
salt:
----------
          ID: nginx
    Function: pkg.installed
      Result: True
     Comment: The following packages were installed/updated: nginx
     Started: 14:56:49.270604
    Duration: 13519.784 ms
     Changes:
              ----------
              nginx:
                  ----------
                  new:
                      1:1.10.2-1.el7
                  old:
              nginx-all-modules:
                  ----------
                  new:
                      1:1.10.2-1.el7
                  old:
              nginx-mod-http-geoip:
                  ----------
                  new:
                      1:1.10.2-1.el7
                  old:
              nginx-mod-http-image-filter:
                  ----------
                  new:
                      1:1.10.2-1.el7
                  old:
              nginx-mod-http-perl:
                  ----------
                  new:
                      1:1.10.2-1.el7
                  old:
              nginx-mod-http-xslt-filter:
                  ----------
                  new:
                      1:1.10.2-1.el7
                  old:
              nginx-mod-mail:
                  ----------
                  new:
                      1:1.10.2-1.el7
                  old:
              nginx-mod-stream:
                  ----------
                  new:
                      1:1.10.2-1.el7
                  old:
----------
          ID: nginx
    Function: service.running
      Result: True
     Comment: Started Service nginx
     Started: 14:57:03.175714
    Duration: 147.686 ms
     Changes:
              ----------
              nginx:
                  True
----------
          ID: nginx
    Function: file.absent
        Name: /etc/nginx/nginx.conf.default
      Result: True
     Comment: Removed file /etc/nginx/nginx.conf.default
     Started: 14:57:03.325230
    Duration: 0.843 ms
     Changes:
              ----------
              removed:
                  /etc/nginx/nginx.conf.default

Summary for salt
------------
Succeeded: 3 (changed=3)
Failed:    0
------------
Total states run:     3
Total run time:  13.668 s
```

**Comment by gtmanfred** [comment]
I have opened the above PR to add call_listen to the state.pkg which will allow for listen and listen_in to be used in salt-ssh. #39904

**Comment by rallytime** [comment]
This was fixed by #39907. Closing.
